langwatch/src/components/prompt-configs/VersionHistoryListPopover.tsx

```diff
@@ -0,0 +1,279 @@
+          </HStack>
+        </VStack>
+        {!isCurrent && (
+          <Tooltip
+            content="Restore this version"
+            positioning={{ placement: "top" }}
+          >
+            <Button variant="ghost" onClick={onRestore} loading={isLoading}>
+              <HistoryIcon size={24} />
+            </Button>
+          </Tooltip>
+        )}
+      </HStack>
+    </VStack>
+  );
+}
+
+/**
+ * Scrollable list of version history items
+ */
+function VersionHistoryList({
+  versions,
+  onRestore,
+  isLoading,
+}: {
+  versions: PromptVersion[];
+  onRestore: (versionId: string) => void;
+  isLoading: boolean;
+}) {
+  return (
+    <VStack
+      align="start"
+      width="full"
+      padding={5}
+      maxHeight="350px"
+      overflowY="auto"
+    >
+      {versions.map((version, index) => (
+        <VersionHistoryItem
+          key={version.id}
+          version={version}
+          onRestore={() => onRestore(version.id)}
+          isCurrent={index === 0}
+          isLoading={isLoading}
+        />
+      ))}
+    </VStack>
+  );
+}
+
+/**
+ * Trigger button for the version history popover
+ */
+function VersionHistoryTrigger({ onClick }: { onClick?: () => void }) {
+  return (
+    <Popover.Trigger asChild onClick={onClick}>
+      <Button variant="ghost" color="gray.500" size="xs">
+        <HistoryIcon size={16} />
+        <Text>Prompt Version History</Text>
+      </Button>
+    </Popover.Trigger>
+  );
+}
+
+/**
+ * Content of the version history popover
+ */
+function VersionHistoryContent({
+  onRestore,
+  versions,
+  isLoading,
+}: {
+  onRestore: (versionId: string) => void;
+  versions: PromptVersion[];
+  isLoading: boolean;
+}) {
+  return (
+    <Popover.Content width="500px">
+      <Popover.Arrow />
+      <Popover.Header fontWeight={600} fontSize="16px">
+        Prompt Version History
+      </Popover.Header>
+      <Popover.CloseTrigger />
+      <Popover.Body padding={0}>
+        <VersionHistoryList
+          versions={versions}
+          onRestore={onRestore}
+          isLoading={isLoading}
+        />
+      </Popover.Body>
+    </Popover.Content>
+  );
+}
+
+/**
+ * Base popover component without API dependencies
+ */
+function VersionHistoryPopover({
+  isOpen,
+  onOpenChange,
+  onRestore,
+  versions,
+  isLoading,
+}: {
+  isOpen: boolean;
+  onOpenChange: (open: boolean) => void;
+  onRestore: (versionId: string) => void;
+  versions: PromptVersion[];
+  isLoading: boolean;
+}) {
+  return (
+    <Popover.Root open={isOpen} onOpenChange={({ open }) => onOpenChange(open)}>
+      <VersionHistoryTrigger />
+      {isOpen && (
+        <VersionHistoryContent
+          onRestore={onRestore}
+          versions={versions}
+          isLoading={isLoading}
+        />
+      )}
+    </Popover.Root>
+  );
+}
+
+/**
+ * Fully composed version history popover with API integration
+ */
+export function VersionHistoryListPopover({ configId }: { configId: string }) {
```

Comment:
We might want to consider only making this fetch call when the list is opened. Right now we're also refetching on open, but there's a better way to do this.

Reply:
Additionally, this file should probably be reconsidered a bit about how it works to make it more usable.